unic/ucd/normal/src/canonical_combining_class.rs

```diff
@@ -53,7 +53,7 @@ impl TotalCharProperty for CanonicalCombiningClass {
 impl NumericCharProperty<u8> for CanonicalCombiningClass {
     /// Get numeric value for character property value
     fn number(&self) -> u8 {
-        self.number()
+        Self::number(self)
```

Comment:
If a type has a method, that always takes precedence. So, calling into such a method from a trait is safe, as long as you remember to implement the function. With the right lint warn enabled, we can easily catch the problem in compile time, if happens. (Which actually did while I was expanding the contracts! :D)